tests/test_group.py

```diff
@@ -428,6 +428,28 @@ def test_group_len(store: Store, zarr_format: ZarrFormat) -> None:
     assert len(group) == 0
 
 
+def test_group_with_context_manager(store: Store, zarr_format: ZarrFormat, overwrite: bool) -> None:
+    spath = StorePath(store)
+
+    # attempt to open a group that does not exist.
+    with pytest.raises(FileNotFoundError):
+        with Group.open(store) as store:
+            pass
+
+    attrs = {"path": "foo"}
+
+    with Group.from_store(
+        store, attributes=attrs, zarr_format=zarr_format, overwrite=overwrite
+    ) as group:
+        assert store._is_open
+        assert group.attrs == attrs
+        assert group.metadata.zarr_format == zarr_format
+        assert group.store_path == spath
+
+    # Check if store was closed after exit.
+    assert not store._is_open
```

Comment:
rather than using this private store api, let's try to just to use the group again and make sure it fails.


```suggestion
    with pytest.raises(ValueError):
        group.attrs['path'] = 'bar'
```

Reply:
@jhamman apologise for such late response on this.
```
with pytest.raises(ValueError):
        group.attrs['path'] = 'bar'
```
does not raise any value error. the store is closed but the group instance still works.

Reply:
If you can edit the Group metadata after the Group has been closed, what is the point of the context manager?